### PR TITLE
Annotation work /File Cleaning part 3

### PR DESCRIPTION
**Pull request by MrRowey**
This Tackles the following files
- PrefetchUtilities.lua
- Profiler.lua
- Prop.lua
- ScenarioUtilites.lua
- ScriptTask.lua
- StrategyBuilder.lua
- Strategymanager.lua
- VizMarker.lua
- score.lua
- simuistate.lua

### CODE REVIEW

File: lua/sim/PrefetchUtilities.lua [hunk 1]
```diff
@@ -6,74 +6,84 @@
 ----                                                                             
 ----  Copyright © 2005 Gas Powered Games, Inc.  All rights reserved.          
 
+local TableGetn = table.getn
+
+---@param armyName string
+---@return any
```
Comment (Hdt80bro) [inline lua/sim/PrefetchUtilities.lua:12]: This is probably the same type as https://github.com/FAForever/fa/blob/deploy/fafdevelop/engine/Core/CPrefetchSet.lua#L5, so we should probably make that a class too

File: lua/sim/Profiler.lua [hunk 2]
```diff
@@ -1,6 +1,5 @@
-
 -- Useful sources to read:
```
Comment (Hdt80bro) [inline lua/sim/Profiler.lua:1]: Sorry, I didn't realize that you'd make it this far! I'm covering this file as part of https://github.com/FAForever/fa/pull/4087 so you didn't have to worry about it ;)

File: lua/sim/Projectile.lua [hunk 3]
```diff
@@ -193,6 +199,11 @@ Projectile = Class(moho.projectile_methods) {
     end,
 
     --- Called by the engine when the projectile receives damage
+    ---@param self Projectile
+    ---@param instigator any
+    ---@param amount integer
+    ---@param vector Vector
+    ---@param damageType string
```
Comment (Hdt80bro) [inline lua/sim/Projectile.lua:206]: I think instigator can be any `Unit` and damageType can be `DamageType`

File: lua/sim/Projectile.lua [hunk 4]
```diff
@@ -507,6 +529,11 @@ Projectile = Class(moho.projectile_methods) {
     -- @param DamageData The damage data passed by the weapon
     -- @param targetEntity The entity we hit, is nil if we hit terrain
     -- @param cachedPosition A cached position that is passed to prevent table allocations, can not be used in fork threads and / or after a yield statement
```
Comment (Hdt80bro) [inline lua/sim/Projectile.lua:531]: Whoops, we forget to delete these extra `@param`s!

File: lua/sim/Projectile.lua [hunk 5]
```diff
@@ -709,6 +745,10 @@ Projectile = Class(moho.projectile_methods) {
     end,
 
     --- Called by Lua to create the impact effects
+    ---@param self Projectile
+    ---@param army integer
+    ---@param EffectTable table
```
Comment (Hdt80bro) [inline lua/sim/Projectile.lua:750]: effect tables are all `string[]`

File: lua/sim/Projectile.lua [hunk 6]
```diff
@@ -782,18 +836,26 @@ Projectile = Class(moho.projectile_methods) {
 
     -- Deprecated functionality
 
+    ---@param self Projectile
+    ---@return any
```
Comment (Hdt80bro) [inline lua/sim/Projectile.lua:840]: I've just been using `Vector` for positions

File: lua/sim/Prop.lua [hunk 7]
```diff
@@ -483,3 +478,9 @@ Prop = Class(moho.prop_methods) {
         return self.CanBeKilled
     end,
 }
+
+
+-- imports kept for backwards compatibility with mods
+local Entity = import('/lua/sim/Entity.lua').Entity
+local DeprecatedWarnings = { }
+local TrashAdd = TrashBag.Add
```
Comment (Hdt80bro) [inline lua/sim/Prop.lua:486]: Nice addition!

**Review by Hdt80bro — COMMENTED** [review]
Didn't quite finish, but I hope the feedback is useful! Again, fantastic job with the annotations Rowey!

File: lua/sim/PrefetchUtilities.lua [hunk 8]
```diff
@@ -6,74 +6,84 @@
 ----                                                                             
 ----  Copyright © 2005 Gas Powered Games, Inc.  All rights reserved.          
 
+local TableGetn = table.getn
```
Comment (Garanas) [inline lua/sim/PrefetchUtilities.lua:9]: I wasn't aware this file existed.

File: lua/sim/Projectile.lua [hunk 9]
```diff
@@ -736,6 +780,11 @@ Projectile = Class(moho.projectile_methods) {
     end,
 
     --- Called by Lua to retrieve the terrain effects
+    ---@param self Projectile
+    ---@param TargetType string
+    ---@param ImpactEffectType string
+    ---@param position number
```
Comment (Garanas) [inline lua/sim/Projectile.lua:786]: Vector

File: lua/sim/Projectile.lua [hunk 10]
```diff
@@ -782,18 +836,26 @@ Projectile = Class(moho.projectile_methods) {
 
     -- Deprecated functionality
```
Comment (Garanas) [inline lua/sim/Projectile.lua:837]: Add ---@deprecated to all the functions below

Comment (Garanas) [reply lua/sim/Projectile.lua:837]: of the Projectile class, that is

File: lua/sim/ScenarioUtilities.lua [hunk 11]
```diff
@@ -54,10 +59,14 @@ function GetMarkers()
...
 end
 
+---@param name string
+---@return any
 function GetMarker(name)
     return Scenario.MasterChain._MASTERCHAIN_.Markers[name]
 end
 
+---@param chainName string
+---@return table
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:69]: Vector[]

File: lua/sim/ScenarioUtilities.lua [hunk 12]
```diff
@@ -84,6 +94,8 @@ function FindParentChain(markerName)
     return nil
 end
 
+---@param name string
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:97]: MarkerChain

File: lua/sim/ScenarioUtilities.lua [hunk 13]
```diff
@@ -92,9 +104,10 @@ function GetMarkerChain(name)
     return chain
 end
 
-----[  MarkerToPosition                                                           ]--
-----[                                                                             ]--
-----[  Converts a marker as specified in *_save.lua file to a position.           ]--
+---MarkerToPosition
+---Converts a marker as specified in *_save.lua file to a position.
+---@param strMarker any
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:109]: Marker

File: lua/sim/ScenarioUtilities.lua [hunk 14]
```diff
@@ -92,9 +104,10 @@ function GetMarkerChain(name)
...
+---@return any
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:110]: Vector

File: lua/sim/ScenarioUtilities.lua [hunk 15]
```diff
@@ -236,9 +263,11 @@ function CreateArmyUnit(strArmy,strUnit)
...
 end
 
-----[  FindUnitGroup                                                              ]--
-----[                                                                             ]--
-----[  Finds the unit group with the specified name.                              ]--
+---FindUnitGroup
+---Finds the unit group with the specified name.
+---@param strGroup any
+---@param tblNode any
+---@return nil
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:270]: table of unit entries (of the save file), so technically just `table` as that can be a lot of things

File: lua/sim/ScenarioUtilities.lua [hunk 16]
```diff
@@ -262,16 +291,23 @@ function FindUnitGroup(strGroup,tblNode)
...
 
-----[  CreateArmySubGroup                                                                      ]--
-----[                                                                                          ]--
-----[  Creates Army groups from a number of groups specified in order from the Units Hierarchy ]--
+---CreateArmySubGroup                                                                                        ]--
+---Creates Army groups from a number of groups specified in order from the Units Hierarchy
+---@param strArmy any
+---@param strGroup any
+---@param ... any
+---@return table
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:308]: first result is a list of units -> `Unit[]`

File: lua/sim/ScenarioUtilities.lua [hunk 17]
```diff
@@ -301,9 +337,15 @@ function CreateArmySubGroup(strArmy,strGroup,...)
...
-----[  CreateSubGroup                                                                      ]--
-----[                                                                                      ]--
-----[  Used by CreateArmySubGroup                                                          ]--
+---CreateSubGroup
+---Used by CreateArmySubGroup
+---@param tblNode any
+---@param strArmy any
+---@param strGroup any
+---@param ... any
+---@return table|any
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:346]: First one is an array of units -> `Unit[]`

File: lua/sim/ScenarioUtilities.lua [hunk 18]
```diff
@@ -322,8 +364,11 @@ function CreateSubGroup(tblNode, strArmy, strGroup, ...)
     return platoonList, tblResult, treeResult
 end
 
-----[  CreateInitialArmyGroup                                                     ]--
-----[                                                                             ]--
+---CreateInitialArmyGroup
+---@param strArmy any
+---@param createCommander Unit
+---@return unknown|nil
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:370]: `Unit[]`

File: lua/sim/ScenarioUtilities.lua [hunk 19]
```diff
@@ -683,19 +727,25 @@ function InitializeScenarioArmies()
...
 end
 
-
-----[ SpawnPlatoon                                                                ]--
-----[ Spawns unit group and assigns to platoon it is a part of                    ]--
+---SpawnPlatoon
+---Spawns unit group and assigns to platoon it is a part of
+---@param strArmy any
+---@param strGroup any
+---@return boolean
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:745]: Platoon

File: lua/sim/ScenarioUtilities.lua [hunk 20]
```diff
@@ -683,19 +727,25 @@ function InitializeScenarioArmies()
...
+---@return any
+---@return any
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:747]: Unit[]

File: lua/sim/ScenarioUtilities.lua [hunk 21]
```diff
@@ -722,6 +772,11 @@ function SpawnPlatoon(strArmy, strGroup)
     return platoonList[platoonName], platoonList, tblResult, treeResult
 end
 
+---@param strArmy any
+---@param strGroup any
+---@return any
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:777]: Platoon[]

File: lua/sim/ScenarioUtilities.lua [hunk 22]
```diff
@@ -722,6 +772,11 @@ function SpawnPlatoon(strArmy, strGroup)
...
+---@return any
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:778]: Unit[]

File: lua/sim/ScenarioUtilities.lua [hunk 23]
```diff
@@ -754,6 +811,16 @@ function CountChildUnits(tblNode)
...
 end
 
+---@param strArmy integer
+---@param tblNode any
+---@param tblResult any
+---@param platoonList any
+---@param currPlatoon Platoon
+---@param treeResult any
+---@param balance number
+---@return nil
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:821]: Platoon[]

File: lua/sim/ScenarioUtilities.lua [hunk 24]
```diff
@@ -754,6 +811,16 @@ function CountChildUnits(tblNode)
...
+---@return any
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:822]: Unit[] (I think)

File: lua/sim/ScenarioUtilities.lua [hunk 25]
```diff
@@ -879,11 +946,15 @@ function CreatePlatoons(strArmy, tblNode, tblResult, platoonList, currPlatoon, t
...
-----[  CreateArmyGroup                                                            ]--
-----[                                                                             ]--
-----[  Creates the specified group in game.                                       ]--
+---CreateArmyGroup
+---Creates the specified group in game.
+---@param strArmy any
+---@param strGroup any
+---@param wreckage Prop
+---@param balance number
+---@return any
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:955]: Unit[]

File: lua/sim/ScenarioUtilities.lua [hunk 26]
```diff
@@ -908,15 +979,18 @@ function CreateArmyGroup(strArmy,strGroup,wreckage, balance)
 end
 
 -- CreateArmyTree
---
 -- Returns tree of units created by the editor. 2nd return is table of units
+---@param strArmy any
+---@param strGroup any
+---@return any
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:985]: Unit[]

File: lua/sim/ScenarioUtilities.lua [hunk 27]
```diff
@@ -935,8 +1012,14 @@ function CreateArmyGroupAsPlatoonBalanced(strArmy, strGroup, formation, OnFinish
...
 -- CreateArmyGroupAsPlatoon
---
 -- Returns a platoon that is created out of all units in a group and its sub groups.
+---@param strArmy any
+---@param strGroup any
+---@param formation any
+---@param tblNode any
+---@param platoon Platoon
+---@param balance any
+---@return any
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:1022]: Platoon

File: lua/sim/ScenarioUtilities.lua [hunk 28]
```diff
@@ -935,8 +1012,14 @@ function CreateArmyGroupAsPlatoonBalanced(strArmy, strGroup, formation, OnFinish
 end
 
 -- CreateArmyGroupAsPlatoon
---
 -- Returns a platoon that is created out of all units in a group and its sub groups.
+---@param strArmy any
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:1016]: string

File: lua/sim/ScenarioUtilities.lua [hunk 29]
```diff
@@ -935,8 +1012,14 @@ function CreateArmyGroupAsPlatoonBalanced(strArmy, strGroup, formation, OnFinish
...
+---@param strGroup any
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:1017]: string

File: lua/sim/ScenarioUtilities.lua [hunk 30]
```diff
@@ -997,6 +1080,11 @@ function CreateArmyGroupAsPlatoon(strArmy, strGroup, formation, tblNode, platoon
 end
 
 -- Creates an army group at a certain veteran level
+---@param strArmy any
+---@param strGroup any
+---@param formation any
+---@param veteranLevel integer
+---@return any
```
Comment (Garanas) [inline lua/sim/ScenarioUtilities.lua:1087]: platoon

File: lua/sim/ScenarioUtilities.lua (hunk 29 above)
Comment (Garanas) [reply lua/sim/ScenarioUtilities.lua:1017]: same for other references

File: lua/sim/ScenarioUtilities.lua (hunk 28 above)
Comment (Garanas) [reply lua/sim/ScenarioUtilities.lua:1016]: same for other references

File: lua/sim/StrategyBuilder.lua [hunk 31]
```diff
@@ -41,45 +46,60 @@ StrategyBuilder = Class(Builder) {
         end
     end,
 
+    ---@param self StrategyBuilder
+    ---@return any
```
Comment (Garanas) [inline lua/sim/StrategyBuilder.lua:50]: boolean

File: lua/sim/StrategyManager.lua [hunk 32]
```diff
@@ -8,16 +8,16 @@
...
---local StrategyList = import('/lua/ai/SkirmishStrategyList.lua').StrategyList
-local AIAddBuilderTable = import('/lua/ai/AIAddBuilderTable.lua')
-local SUtils = import('/lua/AI/sorianutilities.lua')
 
 ---@class StrategyManager : BuilderManager
 StrategyManager = Class(BuilderManager) {
+    ---@param self StrategyManager
+    ---@param brain AIBrain
+    ---@param lType any
+    ---@param location number
```
Comment (Garanas) [inline lua/sim/StrategyManager.lua:18]: Vector

File: lua/sim/StrategyManager.lua [hunk 33]
```diff
@@ -8,16 +8,16 @@
...
+    ---@param radius number
+    ---@param useCenterPoint any
```
Comment (Garanas) [inline lua/sim/StrategyManager.lua:20]: boolean

File: lua/sim/StrategyManager.lua [hunk 34]
```diff
@@ -101,8 +114,23 @@ StrategyManager = Class(BuilderManager) {
     end,
 }
 
+---@param brain AIBrain
+---@param lType any
+---@param location number
```
Comment (Garanas) [inline lua/sim/StrategyManager.lua:119]: vector

File: lua/sim/StrategyManager.lua [hunk 35]
```diff
@@ -101,8 +114,23 @@ StrategyManager = Class(BuilderManager) {
...
+---@param radius number
+---@param useCenterPoint any
```
Comment (Garanas) [inline lua/sim/StrategyManager.lua:121]: boolean

File: lua/sim/VizMarker.lua [hunk 36]
```diff
@@ -21,6 +21,8 @@ local Entity = import('/lua/sim/Entity.lua').Entity
 ---@field WaterVision boolean
 ---@deprecated
 VizMarker = Class(Entity) {
+    ---@param self VizMarker
+    ---@param spec any
```
Comment (Garanas) [inline lua/sim/VizMarker.lua:25]: can you define the spec too? You can base it off what is saved in self

**Review by Garanas — CHANGES_REQUESTED** [review]
@Hdt80bro feel free to merge as you see fit after the release

**Pull request merged**